blockchain/state/statedb.go

```diff
@@ -1092,3 +1103,71 @@ func (s *StateDB) GetContractStorageRoot(contractAddr common.Address) (common.Ha
 	}
 	return contract.GetStorageRoot(), nil
 }
+
+// PrepareAccessList handles the preparatory steps for executing a state transition with
+// regards to EIP-2929:
+//
+// - Add sender to access list (2929)
+// - Add destination to access list (2929)
+// - Add precompiles to access list (2929)
+//
+// This method should only be called if Yolov3/Berlin/2929+2930 is applicable at the current number.
+func (s *StateDB) PrepareAccessList(sender common.Address, feepayer common.Address, dst *common.Address, precompiles []common.Address) {
+	// Clear out any leftover from previous executions
+	s.accessList = newAccessList()
+
+	s.AddAddressToAccessList(sender)
+	if common.EmptyAddress(feepayer) && sender != feepayer {
```

Comment:
Do you want to check which feepayer is not empty?
```suggestion
	if !common.EmptyAddress(feepayer) && sender != feepayer {
```

Reply:
Wow, I didn't noticed it.. Thanks!